src/interface/params.hpp

```diff
@@ -43,10 +43,11 @@ class Params {
   ///
   /// Throws an error if the key is already in use
   template <typename T>
-  void Add(const std::string &key, T value) {
+  void Add(const std::string &key, T value, bool make_mutable = false) {
     PARTHENON_REQUIRE_THROWS(!(hasKey(key)), "Key " + key + " already exists");
     myParams_[key] = std::unique_ptr<Params::base_t>(new object_t<T>(value));
     myTypes_.emplace(make_pair(key, std::type_index(typeid(value))));
+    myMutable_[key] = make_mutable;
```

Comment:
Out of curiosity, is there a reason to use `emplace` in one case and the direct assignment in the other case?
Also, I wonder, if it wouldn't be cleaner the only use one map (e.g., with a struct) instead of three, but that's probably a separate discussion.

Reply:
I used assignment because it was easier. I don't know why the original was `emplace`. I can copy that if it's preferred.